src/app/shared/services/educational-events-rest.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Inject } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+import { SETTINGS, Settings } from '../../settings';
+import { TypeaheadRestService } from './typeahead-rest.service';
+import { EducationalEvent } from '../models/educational-event.model';
+import { switchMap, map } from 'rxjs/operators';
+import { decodeArray } from '../utils/decode';
+import { Observable } from 'rxjs';
+import { DropDownItem } from '../models/drop-down-item.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EducationalEventsRestService extends TypeaheadRestService<
+  typeof EducationalEvent
+> {
+  constructor(http: HttpClient, @Inject(SETTINGS) settings: Settings) {
+    super(http, settings, EducationalEvent, 'EducationalEvents', 'Designation');
+  }
+
+  getTypeaheadItems(term: string): Observable<ReadonlyArray<DropDownItem>> {
+    return this.http
+      .get<unknown>(`${this.baseUrl}/CurrentSemester`, {
+        params: {
+          fields: [this.idAttr, this.labelAttr].join(','),
+          [`filter.${this.labelAttr}`]: `~*${term}*`,
+        },
+      })
+      .pipe(
+        switchMap(decodeArray(this.typeaheadCodec)),
+        map((items) =>
+          items.map((i) => ({ Key: i[this.idAttr], Value: i[this.labelAttr] }))
+        )
+      );
+  }
+
+  getTypeaheadItemById(id: number): Observable<DropDownItem> {
+    return this.http
+      .get<unknown>(`${this.baseUrl}/CurrentSemester`, {
+        params: {
+          fields: [this.idAttr, this.labelAttr].join(','),
+          ['filter.Id']: `=${id}`,
+        },
+      })
+      .pipe(
+        switchMap(decodeArray(this.typeaheadCodec)),
+        map((items) => {
+          if (items.length === 0) {
+            throw new Error('educational event not found');
```

Comment:
Nach mir sollte so die Exception in den `GlobalErrorHandler` kommen und dem User eine Meldung «Ein Fehler ist aufgetreten, versuchen Sie es erneut oder laden Sie die Applikation neu.» angezeigt werden, was nicht ganz schön ist.

Ich würde sowas machen, um das gleiche Verhalten wie im `RestErrorInterceptor` zu erreichen (Error anzeigen und Observable completen):
```
switchMap((items) => {
  if (items.length === 0) {
    this.toastr.error(
      this.translate.instant(`global.rest-errors.notfound-message`),
      this.translate.instant(`global.rest-errors.notfound-title`)
    );
    return EMPTY;
  }
  return of(items[0]);
})
```